tailnet/test/integration/integration.go

```diff
@@ -264,13 +264,18 @@ http {
 
 // StartClientDERP creates a client connection to the server for coordination
 // and creates a tailnet.Conn which will only use DERP to connect to the peer.
-func StartClientDERP(t *testing.T, logger slog.Logger, serverURL *url.URL, myID, peerID uuid.UUID) *tailnet.Conn {
+func StartClientDERP(t *testing.T, logger slog.Logger, serverURL *url.URL, derpMap *tailcfg.DERPMap, clientNumber int, myID, peerID uuid.UUID) *tailnet.Conn {
+	listenPort := uint16(client1Port)
+	if clientNumber == 2 {
+		listenPort = client2Port
+	}
```

Comment:
`clientNumber` feels pretty ambiguous. I think something like `type ClientRole int` would be clearer. From a quick reading it's hard to tell which is which.

Reply:
Refactored into a `Client` struct that contains all the details that gets passed around instead. The client processes still use a "number" so they can select the correct struct as "me", but it's a const now.